lib/ecto/adapters/myxql/connection.ex

```diff
@@ -1443,6 +1459,7 @@ if Code.ensure_loaded?(MyXQL) do
     defp ecto_to_db(:bigserial, _query), do: "bigint unsigned not null auto_increment"
     defp ecto_to_db(:binary_id, _query), do: "binary(16)"
     defp ecto_to_db(:string, _query), do: "varchar"
+    defp ecto_to_db(:bitstring, _query), do: "bit"
```

Comment:
For MySQL it seems more complicated:

- If you use `bit` type in migrations it will default to `bit(1)`. 
- You cannot cast to bit, even if you specify the type (https://dev.mysql.com/doc/refman/8.0/en/cast-functions.html#function_cast)
- There is no such thing as `varbit`

Assuming all the above is correct, maybe we should do this then:

1. Add `defp ecto_size_to_db(:bitstring), do: "bit"`.
2. Raise here `defp ecto_to_db(:bitstring)`

This would mean that migrations work as long as you specify the size and casting raises. This would avoid unexpected surprises in migrations specifying size 1 silently. And since casting doesn't work it would be clearer if we make our own message rather than the user not knowing whether we just sent the wrong type to the database.

Reply:
Done